Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -0,0 +1,411 @@
+  m_exclude_btn->setEnabled(false);
+  m_include_btn->setEnabled(false);
+  std::vector<Diff>().swap(m_include);
+  std::vector<Diff>().swap(m_exclude);
+  JitInterface::SetProfilingState(JitInterface::ProfilingState::Disabled);
+}
+
+void CodeDiffDialog::ClearBlockCache()
+{
+  Core::State old_state = Core::GetState();
+
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Paused);
+
+  JitInterface::ClearCache();
+
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Running);
+}
+
+void CodeDiffDialog::OnRecord(bool enabled)
+{
+  JitInterface::ProfilingState state;
+
+  if (enabled)
+  {
+    ClearBlockCache();
+    m_record_btn->setText(tr("Recording..."));
+    state = JitInterface::ProfilingState::Enabled;
+    m_exclude_btn->setEnabled(true);
+    m_include_btn->setEnabled(true);
+  }
+  else
+  {
+    ClearBlockCache();
+    m_record_btn->setText(tr("Start Recording"));
+    state = JitInterface::ProfilingState::Disabled;
+    m_exclude_btn->setEnabled(false);
+    m_include_btn->setEnabled(false);
+  }
+
+  m_record_btn->update();
+  JitInterface::SetProfilingState(state);
+}
+
+void CodeDiffDialog::OnInclude()
+{
+  std::vector<Diff> current_diff;
+  auto current_symbols = CalculateSymbolsFromProfile();
+  if (m_include.empty() && m_exclude.empty())
+  {
+    m_include = current_symbols;
+    return;
+  }
+
+  if (m_exclude.empty())
+  {
+    current_diff = current_symbols;
+  }
+  else
+  {
+    for (auto& iter : current_symbols)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+      }
+    }
+  }
+
+  if (!m_include.empty())
+  {
+    // Compare include with current and remove items that aren't in both.
+    std::vector<Diff> tmp_swap;
+    for (auto& iter : m_include)
+    {
+      if (std::any_of(current_diff.begin(), current_diff.end(),
+                      [&](Diff& v) { return v.symbol == iter.symbol; }))
+      {
+        tmp_swap.push_back(iter);
+      }
+    }
+    m_include.swap(tmp_swap);
+  }
+  else
+  {
+    m_include = current_symbols;
+    RemoveMatchingSymbolsFromIncludes(m_exclude);
+  }
+}
+
+void CodeDiffDialog::OnExclude()
+{
+  std::vector<Diff> current_diff;
+  auto current_symbols = CalculateSymbolsFromProfile();
+  if (m_include.empty() && m_exclude.empty())
+  {
+    m_exclude = current_symbols;
+    return;
+  }
+
+  if (m_exclude.empty())
+  {
+    m_exclude = current_symbols;
+  }
+  else
+  {
+    for (auto& iter : current_symbols)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+
+        m_exclude.insert(pos, iter);
+      }
+    }
+    // If there is no include list, we're done.
+    if (m_include.empty())
+      return;
+  }
+
+  if (m_exclude.empty())

```

Comment:
Like I eluded to in my other comment, I think this if branch doesn't make a lot of sense.  `RemoveMatchingSymbolsFromInclude` removes the entries from the passed in list.  If there are no entries, there's nothing to remove.

If the logic is working as expected, I guess you can simply remove it:

```
if (!m_exclude.empty())
{
  RemoveMatchingSymbolsFromIncludes(current_diff);
}
```

Reply:
Trying some more testing now w/ break point counter to see if it ever occurs in weird edge cases.

Reply:
Removed. Even if forcing variables to an impossible condition (Qt events executing at the same time) that code will never be hit.